### PR TITLE
Fix failing MongoDB tests

### PR DESCRIPTION
**Pull request by Zlaticanin**
Related PR: https://github.com/hashicorp/vault/pull/16880 

MongoDB tests started failing when Mongo released v6 docker version. I updated MongoDB driver from [mgo](https://github.com/go-mgo/mgo) unmaintained driver to the official [MongoDB driver](https://github.com/mongodb/mongo-go-driver).

I also deleted the MongoDB Standalone Secrets Engine that was causing some of the test failures. We [deprecated](https://developer.hashicorp.com/vault/docs/deprecation) this, but pulled it back to give users more time to migrate. 

Updating the driver and deleting the standalone secrets engine fixed the problem. Results from running tests with the "_latest_" tag:
Screen Shot 2022-10-24 at 10 24 14 AM

### CODE REVIEW

**Comment by Zlaticanin** [comment]
Question for anyone who reviews the PR: Should I include the changelog or add the label?

File: helper/builtinplugins/registry.go [hunk 1]
```diff
@@ -147,15 +146,11 @@ func newRegistry() *registry {
...
 				DeprecationStatus: consts.PendingRemoval,
 			},
-			"consul":     {Factory: logicalConsul.Factory},
-			"gcp":        {Factory: logicalGcp.Factory},
-			"gcpkms":     {Factory: logicalGcpKms.Factory},
-			"kubernetes": {Factory: logicalKube.Factory},
-			"kv":         {Factory: logicalKv.Factory},
-			"mongodb": {
-				Factory:           logicalMongo.Factory,
-				DeprecationStatus: consts.PendingRemoval,
```
Comment (austingebauer) [inline helper/builtinplugins/registry.go:157]: I'm wondering if the right thing to do is move this over to `consts.Removed`. I will check on this to find out.

**Comment by austingebauer** [comment]
I'd say no changelog entry needed for the test fix. Perhaps we'll need a changelog for the `DeprecationStatus` change, but I want to figure out the right thing to do for that first.

File: helper/builtinplugins/registry.go (hunk 1 above)
Comment (tomhjp) [reply helper/builtinplugins/registry.go:157]: Yeah I think `consts.Removed` is correct. Removing this engine seems fine to me, but should probably be separated into its own PR (with a `change` type changelog). The fix part further down this PR may want to be backported, whereas this removal definitely needs to wait for 1.13.0, so having two separate PRs just lets you control the lifecycle of each change more easily.

Comment (austingebauer) [reply helper/builtinplugins/registry.go:157]: +1 @tomhjp - We should separate the removal of the engines into a different PR.

**Pull request closed without merging**